### PR TITLE
Allowing related entities to be referenced by posted parameters

### PR DESCRIPTION
**Pull request by inferont**

### CODE REVIEW

File: Converter/EntityConverter.php [hunk 1]
```diff
@@ -20,6 +20,7 @@
 use Ecentria\Libraries\EcentriaRestBundle\Services\CRUD\CrudTransformer;
 use Doctrine\Common\Persistence\ManagerRegistry;
 use Symfony\Component\Validator\Exception\RuntimeException;
+use Doctrine\ORM\NoResultException;
```
Comment (artempetrovcode) [inline Converter/EntityConverter.php]: I do not see any usages of this.

File: Converter/EntityConverter.php [hunk 2]
```diff
@@ -97,10 +98,7 @@ public function createNewObject($class, Request $request, $create, $options)
         // Convert array into object and test validity
         $object = $this->crudTransformer->arrayToObject($data, $class);
         if ($object instanceof ValidatableInterface && $create) {
-            $violations = $this->crudTransformer->arrayToObjectPropertyValidation($data, $class);
-            $valid = !((bool) $violations->count());
-            $object->setViolations($violations);
-            $object->setValid($valid);
+            $this->validateNewObject($object, $class, $data, $options);
```
Comment (artempetrovcode) [inline Converter/EntityConverter.php:100]: It looks like unit-test for EntityConverter::createNewObject needs to be updated.

Comment (inferont) [inline Converter/EntityConverter.php:100]: Unit test has been added

File: Converter/EntityConverter.php (hunk 1 above)
Comment (inferont) [inline Converter/EntityConverter.php]: Removed

**Comment by inferont** [comment]
Pull request unneeded

**Pull request closed without merging**